app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java

```diff
@@ -343,32 +376,34 @@ public void onComplete(@NonNull Task<DocumentSnapshot> task) {
         });
     }
 
+    /**
+     * Load the dedicated picture of the activity
+     */
     private void loadActivityHeaderPicture() {
-        // Load the dedicated picture of the activity
         activityImage = findViewById(R.id.activity_image_description);
         progressBar = findViewById(R.id.progress_bar_activity_description);
         progressBar.setVisibility(View.VISIBLE);
-        storageReference = FirebaseStorage.getInstance().getReference();
-        if (act != null) {
-            imagePath = "activities/" + act.getActivityId() + "/activityImage.jpg";
-            StorageReference imageRef = storageReference.child(imagePath);
-            imageRef.putFile(uri).addOnCompleteListener(new OnCompleteListener<UploadTask.TaskSnapshot>() {
-                @Override
-                public void onComplete(@NonNull Task<UploadTask.TaskSnapshot> task) {
-                    if (task.isSuccessful()) {
-                        Log.d(TAG, "Get image: SUCCESS");
-                        FirebaseInteraction.getImageFromFirebase(imageRef, activityImage, progressBar);
-                    } else {
-                        Log.d(TAG, "Activity have no image");
-                        activityImage.setImageAlpha(R.drawable.run_woman);
-                    }
+
+        imagePath = "activities/" + activity.getActivityId() + "/activityImage.jpg";
+        StorageReference imageRef = storageReference.child(imagePath);
```

Comment:
I guess we would have to implement the abstraction of Firebase Storage here as well (the one apparently made by Victor). This is also something we would have to do in EditProfileActivity and in ChatActivity (our three Android Activities from which we upload pictures to Firebase Storage).

Reply:
Yes this is a good idea

Reply:
But we will have above all to make sure that this abstraction won't break our app. Because personally I had a bad experience by abstracting Firebase Realtime Database in ChatActivity... We will see, but I personally think that we have to make sure to have a working app at any rate!